eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/OnlyFinalFieldsForImmutableClassCheck.java

```diff
@@ -0,0 +1,102 @@
+// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Verify the classes with annotation @Immutable should have following rules:
+ * <ol>
+ *   <li>Only final fields allowed</li>
+ * </ol>
+ */
+public class OnlyFinalFieldsForImmutableClassCheck extends AbstractCheck {
+    private static final String IMMUTABLE_NOTATION = "Immutable";
+    private static final String ERROR_MSG = "The variable field ''%s'' of class ''%s'' should be final." +
+        "Classes annotated with @Immutable are supposed to be immutable.";
+
+    private static boolean hasImmutableAnnotation;
```

Comment:
why static?

Reply:
no reason. probably a copy/paste issue XD.
Will remove it.
Tks

Reply:
also, I changed javadocs from private methods to simple comments.